File: reverse-integer/rev.cpp
#include<iostream>

// reverse number:
// 1234 to 4321
// approach: use maths!


const int reverse(int& numberToReverse){
    int rev;
    while(numberToReverse > 0){
        int lastDigit = numberToReverse % 10;
        rev = (rev * 10) + lastDigit;
        numberToReverse = numberToReverse / 10;
    }
    return rev;
}




int main(){
    int number{123450};
    std::cout<<reverse(number)<<std::endl;

}

File: reverse-integer/rev.py
""" check cpp for problem statement """

def reverse(number):
    reverse=0
    numberToRev=number
    while(numberToRev > 0):
        lastDigit = numberToRev % 10
        reverse = (reverse * 10) + lastDigit
        numberToRev = int(numberToRev / 10)
        print("number bein reversed " + str(number))
        print("reverse number " + str(reverse))

if __name__ == '__main__':
    reverse(1233450)
